ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/ConnectIntegrationTest.java

```diff
@@ -239,6 +252,68 @@ public void shouldReturnError() {
         containsString("Connector mock-connector already exists"));
   }
 
+  @Test
+  public void shouldReadAndWriteTimestampsToConnect() throws UnsupportedEncodingException {
+    // Given:
+    create("mock-source", ImmutableMap.<String, String> builder()
+        .put("connector.class", "org.apache.kafka.connect.tools.VerifiableSourceConnector")
+        .put("topic", "foo")
+        .put("throughput", "5")
+        .put("id", "123")
+        .put("topic.creation.default.replication.factor", "1")
+        .put("topic.creation.default.partitions", "1")
+        .build());
+    assertThatEventually(
+        () -> ((ConnectorList) ksqlRestClient.makeKsqlRequest("SHOW CONNECTORS;").getResponse()
+            .get(0)).getConnectors().get(0).getState(),
+        is("RUNNING (1/1 tasks RUNNING)"));
+
+    // When:
+    assertThatEventually(
```

Comment:
This assertion must be in the `then` section. It is when we do our checks that everything works. Btw, do you really need to  check the stream was created successfully? I see you have a `SELECT` clause in the Then section, so that assumes the streams is already created, it will fail if not, so you have your check there.

Reply:
This was added because it takes some time for the topic to actually be created, so the command actually fails the first few times. I rewrote this to just keep trying until timeout instead of using an asset.